occurrences/models.py

```diff
@@ -241,7 +241,9 @@ def is_editable_by_user(self, user):
     def cancel(self, reason=None):
         self.cancelled = True
         self.save()
-        for e in self.enrolments.all():
+        for e in self.enrolments.exclude(
+            Q(status=Enrolment.STATUS_CANCELLED) | Q(status=Enrolment.STATUS_DECLINED)
+        ):
```

Comment:
Exclude cancelled and declined enrolments from occurrence cancel handler

Reply:
Should we explicitly say which statuses are ok rather than saying which things are not ok? Just thinking that if someone adds a new status then its by default ok.

Reply:
I was thinking the opposite :smile: I thought it would be better to send the notification by default. And there is a greater chance that the new status is one of the “allowed” ones rather than rejected ones.

Too much communication would be better than too little.